File: 2018/day1/part1.exs
start = System.monotonic_time()

File.stream!("./input.txt")
|> Stream.map(&String.replace(&1, "\n", ""))
|> Stream.map(&Integer.parse(&1))
|> Enum.reduce(0, fn {value, _}, acc ->
  acc + value
end)
|> IO.inspect()

finish = System.monotonic_time()
diff = finish - start
IO.inspect(diff)

File: 2022/day2/part1.mjs
import fs from "fs";

console.log(
  fs
    .readFileSync("./input.txt", "utf8")
    .split("\n")
    .reduce(
      (acc, game) => {
        const match = game.split(' ').map(x => parseInt(x, 36) % 9 % 5);
        return acc + match[1] + (match[1] === match[0] ? 3 : match[0] % 3 + 1 === match[1] ? 6 :  0);
      },
      0
    )
);